py3status/modules/imap.py

```diff
@@ -27,17 +29,24 @@ def check_mail(self, i3s_output_list, i3s_config):
         mail_count = self._get_mail_count()
 
         response = {
-            'cached_until': time() + self.cache_timeout,
-            'full_text': '{}: {}'.format(self.name, mail_count)
+            'cached_until': time() + self.cache_timeout
         }
 
-        new_mail_color = i3s_config['color_good']
-        check_failed_color = i3s_config['color_bad']
+        if not self.new_mail_color:
+            self.new_mail_color = i3s_config['color_bad']
 
         if mail_count == 'N/A':
-            response['color'] = check_failed_color
+            response['color'] = ''
+            response['full_text'] = mail_count
         elif mail_count != 0:
-            response['color'] = new_mail_color
+            response['color'] = self.new_mail_color
+            response['full_text'] = self.name.replace('%unseen', str(mail_count))
```

Comment:
I think that using format() placeholders would be more elegant, plz check https://github.com/ultrabug/py3status/blob/master/py3status/modules/net_rate.py for usage.